### PR TITLE
[AAP-69617] Optimize user list endpoint to eliminate N+1 queries

### PR DESCRIPTION
**Pull request by huffmanca**
UserViewSet.get_queryset() now adds select_related for modified_by, created_by, and last_login_from on list views to prevent lazy-loading FK objects during serialization. Also annotates is_platform_auditor as an Exists subquery instead of running one .exists() call per user.

The select_related adds self-referential JOINs (User -> User) which may affect query plans on very large user tables (100K+ users). The annotation adds one subquery per list request.

These optimizations reduce GET /users/ SQL queries from 181 to 9 and CPU time by 78% for a 50-user page.

Requires https://github.com/ansible/django-ansible-base/pull/992 .

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Test update
- [ ] Refactoring (no functional changes)
- [ ] Development environment change
- [ ] Configuration change

## Self-Review Checklist

- [x] I have performed a self-review of my code
- [x] I have added relevant comments to complex code sections
- [x] I have updated documentation where needed
- [x] I have considered the security impact of these changes
- [x] I have considered performance implications
- [x] I have thought about error handling and edge cases
- [x] I have tested the changes in my local environment

### Steps to Test
1. 
2. 
3. 

### Required Actions

- [ ] Requires documentation updates
  
- [ ] Requires downstream repository changes
  
- [ ] Requires infrastructure/deployment changes
  
- [ ] Requires coordination with other teams
  
- [ ] Blocked by PR/MR: #XXX
  

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved user list performance and query efficiency; list now includes related and summary fields and correctly reflects platform-auditor status using an optimized lookup.
* **Tests**
  * Added performance and regression tests for user list/detail endpoints covering query scaling, response fields, and platform-auditor behavior.

### CODE REVIEW

File: aap_gateway_api/tests/views/api/v1/user/test_user_list_performance.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        With optimization, query count should remain roughly constant."""
+        from aap_gateway_api.models import User
+
+        url = get_relative_url("user-list")
+
+        # Baseline with few users
+        connection.queries_log.clear()
+        response = admin_api_client.get(url)
+        assert response.status_code == 200
+        baseline_queries = len(connection.queries_log)
```
Comment (john-westcott-iv) [inline aap_gateway_api/tests/views/api/v1/user/test_user_list_performance.py]: **Minor [Quality]:** `connection.queries_log` only populates when `settings.DEBUG = True`. Without it, both baselines are 0 and the assertion `0 < 10` passes vacuously without actually measuring anything.

Same issue applies to the summary fields test on line 80.

```suggestion
    @override_settings(DEBUG=True)
    def test_user_list_query_count_scales_sublinearly(self, admin_api_client, system_user):
```

Alternatively, use `CaptureQueriesContext(connection)` which works regardless of DEBUG setting.

**Review by john-westcott-iv — COMMENTED** [review]
## AI Code Review: 9.8/10 — READY_FOR_HUMAN_REVIEW

| Lens | Score | Findings |
|---|---|---|
| Functionality | 10/10 | None |
| Security | 10/10 | None |
| Quality | 9.5/10 | 1 Minor, 2 Nit |
| **Overall** | **9.8/10** | |

The optimization is well-structured — `select_related` + `Exists` annotation is the standard Django approach to N+1 elimination. The annotation is semantically equivalent to the per-object `.exists()` call. `last_login_from` confirmed as a ForeignKey to Authenticator (line 83 of user model). One minor item on test robustness.

**Note:** The `_annotated_is_platform_auditor` attribute trusts presence for platform auditor status. The pre-existing `_is_platform_auditor` has the same pattern, so no new attack surface is introduced, but worth being aware of.

**Dependency:** This PR requires [django-ansible-base#992](https://github.com/ansible/django-ansible-base/pull/992) for the `select_related` optimization in `related_fields()` and `get_summary_fields()`.

<sub>Reviewed with `aap-sdlc-harness:code-review`</sub>

File: aap_gateway_api/tests/views/api/v1/user/test_user_list_performance.py (hunk 1 above)
Comment (huffmanca) [reply aap_gateway_api/tests/views/api/v1/user/test_user_list_performance.py]: Updated to use `CaptureQueriesContext` to be consistent with the DAB changes.

**Review by john-westcott-iv — APPROVED** [review]
Re-review: the previous Minor finding (use `CaptureQueriesContext` instead of `connection.queries_log`) has been addressed. No new issues found. The optimization is clean and well-tested.

**Comment by huffmanca** [comment]
PR recreated at https://github.com/ansible/jewel/pull/24
